zipkin/src/main/java/zipkin/internal/Collector.java

```diff
@@ -55,13 +63,35 @@ protected void acceptSpans(byte[] serializedSpans, D decoder, Callback<Void> cal
     List<S> spans;
     try {
       spans = decodeList(decoder, serializedSpans);
+      spans = filterSpans(spans, callback);
     } catch (RuntimeException e) {
       callback.onError(errorReading(e));
       return;
     }
     accept(spans, callback);
   }
 
+  /**
+   * Take the list of spans and pump them through pre-configured filters
+   *
+   * @param spans
+   */
+  List<S> filterSpans(List<S> spans, Callback<Void> callback) {
+    // These should never be processed in parallel.
+    // If you need parallel processing put that into your specific filter implementation.
+    boolean isV1Span = spanType.getType().equals(Span.class);
```

Comment:
we won't be filtering on two types. you can assume later it will be only zipkin2.span

Reply:
Ahh, that's good news; I won't need the guava token type once that is that case. In terms of "later", do you want me to make this only supported for v2 as part of this PR?